File: src/Z.Net/System.Net.Mail.MailMessage/MailMessage.SendAsync.cs
// Description: C# Extension Methods | Enhance the .NET Framework and .NET Core with over 1000 extension methods.
// Website & Documentation: https://csharp-extension.com/
// Issues: https://github.com/zzzprojects/Z.ExtensionMethods/issues
// License (MIT): https://github.com/zzzprojects/Z.ExtensionMethods/blob/master/LICENSE
// More projects: https://zzzprojects.com/
// Copyright © ZZZ Projects Inc. All rights reserved.
using System.Net.Mail;

public static partial class Extensions
{
    /// <summary>
    ///     A MailMessage extension method that sends this message asynchronous.
    /// </summary>
    /// <param name="this">The @this to act on.</param>
    /// <param name="userToken">The user token.</param>
    public static void SendAsync(this MailMessage @this, object userToken)
    {
        using (var smtpClient = new SmtpClient())
        {
            smtpClient.SendAsync(@this, userToken);
        }
    }
}

File: src (VB.NET)/Z.ExtensionMethods.VB/Z.Net/System.Net.Mail.MailMessage/MailMessage.SendAsync.vb

' Copyright (c) 2015 ZZZ Projects. All rights reserved
' Licensed under MIT License (MIT) (https://github.com/zzzprojects/Z.ExtensionMethods)
' Website: http://www.zzzprojects.com/
' Feedback / Feature Requests / Issues : http://zzzprojects.uservoice.com/forums/283927
' All ZZZ Projects products: Entity Framework Extensions / Bulk Operations / Extension Methods /Icon Library

Imports System.Net.Mail

Public Module Extensions_801

	''' <summary>
	'''     A MailMessage extension method that sends this message asynchronous.
	''' </summary>
	''' <param name="this">The @this to act on.</param>
	''' <param name="userToken">The user token.</param>
	<System.Runtime.CompilerServices.Extension> _
	Public Sub SendAsync(this As MailMessage, userToken As Object)
		Using smtpClient = New SmtpClient()
			smtpClient.SendAsync(this, userToken)
		End Using
	End Sub
End Module


